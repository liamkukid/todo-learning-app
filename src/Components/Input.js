import { useRef } from 'react';
import { useDispatch } from 'react-redux';
import style from './input.module.scss';
import { addTodo } from '../features/todosSlice';

export default function Input() {
  const dispatch = useDispatch();
  const inputRef = useRef();

  const handleSubmit = e => {
    e.preventDefault();
    e.stopPropagation();
    if (inputRef.current.value) {
      dispatch(addTodo(inputRef.current.value));
    }
  };

  return (
    <form className={style.form_content} onSubmit={handleSubmit}>
      <input
        className={style.form_input}
        name="todo_input"
        type="text"
        id="todo_input"
        ref={inputRef}
        placeholder="New todo..."
      />
      <button className={style.form_submit} type="submit">
        Save
      </button>
    </form>
  );
}
