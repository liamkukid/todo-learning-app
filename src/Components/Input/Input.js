/* eslint-disable prefer-arrow-callback */
import { forwardRef } from 'react';
import style from './input.module.scss';

const Input = forwardRef(function Input(prop, ref) {
  return (
    <input
      className={style.input}
      type="text"
      placeholder="New todo..."
      ref={ref}
    />
  );
});

export default Input;
