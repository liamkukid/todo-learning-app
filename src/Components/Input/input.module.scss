@import '../../styles/variables.scss';

.input {
    width: 80vw;
    max-width: 450px;
    height: 26px;
    margin-bottom: 2px;
    border-radius: 20px;
    border: 2px solid var(--navy);
    background-color: var(--grey);
    font-size:20px;
    font-family:'revert';
    text-indent: 6px;
}

.input:focus,
.input:hover {
    outline: none;
    border: 2px solid var(--navy);
    color: var(--navy);
    background-color: var(--light);
}
