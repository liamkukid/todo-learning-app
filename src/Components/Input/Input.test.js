import renderer from 'react-test-renderer';
import Input from './Input';

it('renders correctly for Input', () => {
  const tree = renderer.create(<Input />);
  expect(tree).toMatchSnapshot();
});
