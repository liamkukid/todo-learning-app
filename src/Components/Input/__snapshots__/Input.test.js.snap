// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for Input 1`] = `
<input
  className="input"
  placeholder="New todo..."
  type="text"
/>
`;
